staging/src/k8s.io/apiserver/pkg/cel/library/cost.go

```diff
@@ -48,6 +50,22 @@ var knownUnhandledFunctions = map[string]bool{
 	"strings.quote": true,
 }
 
+// TODO: Replace this with a utility that extracts types from libraries.
+var knownKubernetesRuntimeTypes = sets.New[reflect.Type](
+	reflect.ValueOf(cel.URL{}).Type(),
+	reflect.ValueOf(cel.IP{}).Type(),
+	reflect.ValueOf(cel.CIDR{}).Type(),
+	reflect.ValueOf(&cel.Format{}).Type(),
+	reflect.ValueOf(cel.Quantity{}).Type(),
+)
+var knownKubernetesCompilerTypes = sets.New[ref.Type](
+	cel.CIDRType,
+	cel.IPType,
+	cel.FormatType,
+	cel.QuantityType,
+	cel.URLType,
+)
```

Comment:
@liggitt Options here are to merge this until https://github.com/kubernetes/kubernetes/issues/126365 is solved or drop this commit. WDYT?

Reply:
https://github.com/kubernetes/kubernetes/pull/126368 is the basis of what I'd like to replace this with. It enforces that all the libraries we use in our base environment are registered with types and then we can use that in the cost system to make sure equality is checked.